backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/export/svg/DiagramExportService.java

```diff
@@ -60,8 +64,10 @@ public String export(Diagram diagram) {
 
         svg.append("<g transform=\"scale(1) translate(0,0)\">"); //$NON-NLS-1$ )
 
-        diagram.getNodes().forEach(node -> svg.append(this.nodeExport.export(node)));
-        diagram.getEdges().forEach(edge -> svg.append(this.edgeExport.export(edge)));
+        Map<String, Optional<NodeAndContainerId>> id2NodeHierarchy = new HashMap<>();
```

Comment:
Why not just a `Map<String, NodeAndContainerId>`, and simply use `Map.containsKey()` to test the presence of a particular id?

Reply:
I agree, I'm not a fan of `Map<XXX, Optional<YYY>>` since it does not even prevent the value to be `null`.

Reply:
I've fixed that myself